homeassistant/components/tahoma/config_flow.py

```diff
@@ -0,0 +1,107 @@
+from homeassistant.helpers import config_validation as cv
+
+from .const import CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL, MIN_UPDATE_INTERVAL
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Somfy TaHoma."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Handle the flow."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_validate_input(self, user_input):
+        """Validate user credentials."""
+        username = user_input.get(CONF_USERNAME)
+        password = user_input.get(CONF_PASSWORD)
+
+        async with TahomaClient(username, password) as client:
+            await client.login()
+            return self.async_create_entry(title=username, data=user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step via config flow."""
+        errors = {}
+
+        if user_input:
+            await self.async_set_unique_id(user_input.get(CONF_USERNAME))
+            self._abort_if_unique_id_configured()
+
+            try:
+                return await self.async_validate_input(user_input)
+            except TooManyRequestsException:
+                errors["base"] = "too_many_requests"
+            except BadCredentialsException:
+                errors["base"] = "invalid_auth"
+            except (TimeoutError, ClientError):
+                errors["base"] = "cannot_connect"
+            except MaintenanceException:
+                errors["base"] = "server_in_maintenance"
+            except Exception as exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+                _LOGGER.exception(exception)
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for TaHoma."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+
+        if self.options.get(CONF_UPDATE_INTERVAL) is None:
+            self.options[CONF_UPDATE_INTERVAL] = DEFAULT_UPDATE_INTERVAL
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Somfy TaHoma options."""
+        return await self.async_step_update_interval()
+
+    async def async_step_update_interval(self, user_input=None):
+        """Manage the options regarding interval updates."""
+        if user_input is not None:
+            self.options[CONF_UPDATE_INTERVAL] = user_input[CONF_UPDATE_INTERVAL]
```

Comment:
Not sure if this has any effect. A new options flow instance is created every time we start the options flow.

Reply:
Thanks, I will have a look and make the change if this code isn't required. I added it based on other integrations, but it could be that it isn't required :).

Reply:
Fixed!